my_methods.rb

```diff
@@ -0,0 +1,411 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |num|
+      yield(num)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |num|
+      yield(num, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      arr = []
+      my_each do |num|
+        arr << num if yield(num)
+      end
+      arr
+    elsif is_a?(Hash)
+      hash = {}
+      my_each do |key, val|
+        hash[key] = val if yield(key, val)
+      end
+      hash
+    end
+  end
+
+  def my_all?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return false unless val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false unless val.is_a?(var) }
+      else
+        my_each { |val| return false if val != var }
+      end
+      return true
+    end
+    return true unless block_given?
+
+    my_each do |num|
+      return false unless yield(num)
+    end
+    true
+  end
+
+  def my_any?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return true if val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return true if val.is_a?(var) }
+      else
+        my_each { |val| return true if val == var }
+      end
+      return false
+    end
+    return true unless block_given?
+
+    my_each do |num|
+      return true if yield(num)
+    end
+    false
+  end
+
+  def my_none?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return false if val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false if val.is_a?(var) }
+      else
+        my_each { |val| return false if val == var }
+      end
+      return true
+    end
+    return false unless block_given?
+
+    my_each do |num|
+      return false if yield(num)
+    end
+    true
+  end
+
+  def my_count
+    return size unless block_given?
+
+    i = 0
+    my_each do |num|
+      i += 1 if yield(num)
+    end
+    i
+  end
```

Comment:
`my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- counts the number of items in enum that are equal to item if an argument is given
```ruby
[1, 1, 2 ,3].my_count(1)  #should return 2
```